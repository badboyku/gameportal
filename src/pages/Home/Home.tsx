import { HelloWorld } from '../../components';
import { getEnvVars } from '../../utils/env';
import logoUrl, { ReactComponent as Logo } from './logo.svg';
import './style.scss';
import './styles.css';

type Props = {};

const Home = (_props: Props) => {
  const { REACT_APP_MY_VAR } = getEnvVars();

  return (
    <div>
      <h1>Home</h1>
      <HelloWorld />
      <img src={logoUrl} className="logo" alt="logo" />
      <Logo width={40} />
      <h3>Env Vars</h3>
      <div>
        IS_DEV: <span style={{ fontWeight: 'bold' }}>{IS_DEV.toString()}</span>
      </div>
      <div>
        IS_PROD: <span style={{ fontWeight: 'bold' }}>{IS_PROD.toString()}</span>
      </div>
      <div>
        REACT_APP_MY_ENVVAR: <span style={{ fontWeight: 'bold' }}>{REACT_APP_MY_VAR}</span>
      </div>
    </div>
  );
};

export default Home;
