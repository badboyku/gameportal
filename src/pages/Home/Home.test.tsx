import { render, screen } from '@testing-library/react';
import Home from './Home';

jest.mock('../../components/HelloWorld', () => ({ HelloWorld: () => <div data-testid="helloWorld" /> }));

const renderHomePage = (props = {}) => {
  return render(<Home {...props} />);
};

describe('Component HomePage', () => {
  it('renders without crashing', () => {
    const { asFragment } = renderHomePage();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders header Home', () => {
    renderHomePage();

    expect(screen.getByText('Home')).toBeInTheDocument();
  });

  it('renders HelloWorld Component', () => {
    renderHomePage();

    expect(screen.getByTestId('helloWorld')).toBeInTheDocument();
  });
});
