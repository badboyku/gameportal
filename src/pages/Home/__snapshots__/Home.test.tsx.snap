// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component HomePage renders without crashing 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Home
    </h1>
    <div
      data-testid="helloWorld"
    />
    <img
      alt="logo"
      class="logo"
      src="logo.svg"
    />
    <svg
      width="40"
    >
      logo.svg
    </svg>
    <h3>
      Env Vars
    </h3>
    <div>
      IS_DEV: 
      <span
        style="font-weight: bold;"
      >
        false
      </span>
    </div>
    <div>
      IS_PROD: 
      <span
        style="font-weight: bold;"
      >
        true
      </span>
    </div>
    <div>
      REACT_APP_MY_ENVVAR: 
      <span
        style="font-weight: bold;"
      />
    </div>
  </div>
</DocumentFragment>
`;
